### ISSUE TITLE
Cannot run NUnit Console from another directory

### ISSUE THREAD

**Issue by rprouse** [nunit/nunit#853]
This has crept in recently, possibly with the new extensions? I am trying to run nunit-console from the root of the source tree,

```
λ .\bin\Debug\nunit-console.exe .\bin\Debug\net-2.0\nunit.framework.tests.dll
NUnit Console Runner 3.0.5751 (Debug)
Copyright (C) 2014 Charlie Poole

Failed to resolve assembly: 'nunit.engine.api, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'
```

**Comment by CharliePoole** [comment]
I see the same thing, so it's not just you. :-)

I can run without arguments, giving the help display, but any assembly argument gives that result. That includes 'junk.dll'.

**Comment by CharliePoole** [comment]
It is the ExtensionService. When initializing, it throws at CecilExtensions.cs, line 53, which is the first point where Resolve is called on a type reference. This method converts the reference to a full-fledged type description, which requires Cecil to locate all referenced assemblies. I'll have to read up on how Cecil searches for assemblies.

**Comment by roboter** [comment]
I have same issue in just downloaded nunit3-beta5

**Comment by CharliePoole** [comment]
@roboter We changed code and fixed the issue a few weeks ago, so it's unlikely that it's actually the same cause. But if you can give a description of how to reproduce this, we can reopen this or start a new issue.

**Comment by roboter** [comment]
Steps to reproduce
I created new project in VisualStudio2015
add Nunit 3.0beta-5 to nuget 
project available on github https://github.com/roboter/NUnit3.0DemoPtoject
run all tests from studio - results ok

create bat file (available in github too) 
C:\"Program Files (x86)"\nunit3-beta-5\bin\nunit-console.exe
C:\"Program Files (x86)"\nunit3-beta-5\bin\nunit-console.exe [full Path to dll]\UnitTestDemo.dll
 pause

first nunit-console without any parameters - display help 
second run 

NUnit Console Runner 3.0.5767
Copyright (C) 2014 Charlie Poole

Failed to resolve assembly: 'nunit.engine.api, Version=3.0.0.0, Culture=neutral,
 PublicKeyToken=2638cd05610744eb'

**Comment by cwesterlund** [comment]
Got the same issue today when installing NUnit 3.0 beta5 from nunit.org

**Comment by NikolayPianikov** [comment]
As workaround you could copy nunit.engine.api.dll to the "addins" directory

**Comment by CharliePoole** [comment]
Workaround is good. Problem is with Cecil trying to load the api assembly rather than the actual .NET load.

**Comment by Guson** [comment]
Hi!
Trying out the new _NUnit_ `3`. Previously running the `2.6.3` version. I did the following steps:
1. Copy the `nunit.engine.api.dl` to the `nunit.org\bin\addins`.
2. Changed the command parameters from `/` to `--`.
3. Removed old command parameters, i.e. `nodots`.
4. Updated the batch command in my Jenkins, see below.
5. Investigating error in Jenkins module _Publish NUnit test result report_, looking for an update of the plugin.

```batch
@echo off
:: Windows cmd-file
:: Runs all debug unit tests for all NUnit-project files below current directory.
:: NUnit-Console | http://nunit.org/index.php?p=nunit-console&r=3.0

:: Assume Unit test runner in path
where /q nunit-console.exe || for %%l in (echo goto:end) do %%l ERROR: Unit test runner program 'nunit-console.exe' cannot be found.

:test
echo Processing Unit test runner, please wait...
dir /b /s *.nunit | find /i /v "\tags\" | find /i /v "\branches\" | find /i /v "\packages\" > tests.txt
for /f "tokens=*" %%i in ('type tests.txt') do (
    echo Processing: %%~nxi
    nunit-console "%%i" --result="%%~dpiUnitTestResults.xml" --config:Debug
)

echo Unit test done.
:end
```

**Comment by rprouse** [comment]
@Guson, you cannot just copy nunit.engine.api.dll into the old addins directory of 2.6.x. You must use a full install of v3 of nunit-console including the new engine and update your batch file accordingly.

If you want it to work with your batch file, you will need to uninstall 2.6.3 or remove it from your path, install v3 of the console and add it to your path.

For point 2, / will work for command line parameters on Windows. -- is better because it is cross platform.

**Comment by Guson** [comment]
It seems that you can force the output into nunit2 schema, by adding `;format=nunit2`, change to the following line, from post above: 
```batch
nunit-console "%%i" --result="%%~dpiUnitTestResults.xml";format=nunit2 --config:Debug
```

**Comment by Guson** [comment]
@rprouse I have a new computer and only installed the NUnit 3 beta5.
Okey, I will use the `--` further more :smiley:

**Comment by CharliePoole** [comment]
@Guson Note that your execution loop defeats NUnit's ability to run the various assemblies in parallel.

Of course, you may need to do this, if Jenkins cannot handle parallel execution in tests or in order to have separate result files per assembly.

However, if those factors do not apply, you may get better performance by building a single command-line with all the assemblies in it and using the --agents option to limit the number of processes that NUnit will run in parallel.

**Comment by Guson** [comment]
@CharliePoole Okey, good point, in my `nunit`-files I have several test-dll's (2-10).
The performance is of no concern at the moment, not so many test in that repository (~1000), but we are adding test so in the future, I'll add a `nunit`-file in the root that include all the others below.
Thanks for the tip.
